File: 351004/Romanovskiy/task361/publisher/src/main/java/com/example/task361/domain/entity/Author.java
package com.example.task361.domain.entity;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;
import com.example.task361.security.Role;

@Entity
@Table(name = "tbl_author")
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class Author extends BaseEntity {
    @Column(length = 64, unique = true, nullable = false)
    private String login;

    @Column(length = 128, nullable = false)
    private String password;

    @Enumerated(EnumType.STRING)
    @Column(length = 16, nullable = false)
    private Role role;

    @Column(name = "firstname", length = 64)
    private String firstname;

    @Column(name = "lastname", length = 64)
    private String lastname;
}

File: 351004/Kuchko/ServerApp/ServerApp/Models/Entities/Author.cs
namespace ServerApp.Models.Entities;

public class Author : BaseEntity
{
    public string Login { get; set; } = null!;
    public string Password { get; set; } = null!;
    public string Firstname { get; set; } = null!;
    public string Lastname { get; set; } = null!;
}